homeassistant/components/opentherm_gw/climate.py

```diff
@@ -258,9 +275,14 @@ async def async_set_temperature(self, **kwargs):
             temp = float(kwargs[ATTR_TEMPERATURE])
             if temp == self.target_temperature:
                 return
-            self._new_target_temperature = await self._gateway.gateway.set_target_temp(
-                temp
-            )
+            if self.setpoint_ovrd_mode == CONF_OVRD_MODE_TEMPORARY:
+                self._new_target_temperature = (
+                    await self._gateway.gateway.set_target_temp(temp, True)
+                )
+            if self.setpoint_ovrd_mode == CONF_OVRD_MODE_CONSTANT:
+                self._new_target_temperature = (
+                    await self._gateway.gateway.set_target_temp(temp, False)
+                )
```

Comment:
As I understand it, both options are mutually exclusive. If it doesn't match the first, there's no need to check the second. I would suggest to refactor `CONF_SETPOINT_OVRD_MODE` to a boolean value, then there's no need for any `CONF_OVRD_MODE_*` vars and we can just do
```
self._new_target_temperature = (
    await self._gateway.gateway.set_target_temp(temp, self.setpoint_ovrd_mode)
)
```

Reply:
What you are saying is correct. I was planning on doing a boolean at first, but changed my mind because I thought this would be easier to understand for users from the options panel. If you really suggest to use a boolean I will revert the changes for this PR and add the boolean option in a separate PR (splitting the bullets as Martin said)

Reply:
It should be safe to assume that users of the gateway are familiar with the terms `temporary` and/or `constant` in relation to temperature overrides (otmonitor uses only these terms). Having a checkbox with one of these terms in the options panel should be clear enough, especially when it is further explained in the docs.
Another option would be to keep the dropdown in the options panel and deriving a boolean from its value upon submission of the form, but in my opinion the checkbox would be sufficient.

Reply:
Suppose you're right on this one. I will change the option to a boolean, will try to do so in a separate PR.
Would mean I have to update documentation and tests as well but that should be fine.

What do you reckon on the schema in __init__.py, should it be removed now? Or should I just remove the parts I modified?